File: src/main/java/com/bridgelabz/employeepayrollappdevelopment/util/EmailListener.java
package com.bridgelabz.employeepayrollappdevelopment.util;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;


@Component
public class EmailListener {

    @Autowired
    private EmailSenderService sender;

//    @EventListener(ApplicationReadyEvent.class)
    public void sendMail() {
        sender.sendEmail("neeta.mourya@bridgelabz.com", "Test Email", "Registered SuccessFully1");


    }
}

File: src/main/java/com/bridgelabz/employeepayrollappdevelopment/util/EmailListener.kt
package com.bridgelabz.employeepayrollappdevelopment.util;
import com.bridgelabz.employeepayrollappdevelopment.util.EmailSenderService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class EmailListener {
    @Autowired
    private val sender: EmailSenderService? = null

    //@EventListener(ApplicationReadyEvent.class)
    fun sendMail() {
        sender!!.sendEmail("neeta.mourya@bridgelabz.com", "Test Email", "Registered SuccessFully1")
    }
}